libs/barista-components/slider/src/slider.ts

```diff
@@ -0,0 +1,408 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DOWN_ARROW,
+  END,
+  HOME,
+  LEFT_ARROW,
+  PAGE_DOWN,
+  PAGE_UP,
+  RIGHT_ARROW,
+  UP_ARROW,
+} from '@angular/cdk/keycodes';
+import { Platform } from '@angular/cdk/platform';
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Input,
+  NgZone,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import { clamp, isDefined } from '@dynatrace/barista-components/core';
+import { DtInput } from '@dynatrace/barista-components/input';
+import {
+  BehaviorSubject,
+  fromEvent,
+  merge,
+  Observable,
+  of,
+  Subject,
+  animationFrameScheduler,
+} from 'rxjs';
+import {
+  debounceTime,
+  distinctUntilChanged,
+  filter,
+  map,
+  share,
+  switchMap,
+  takeUntil,
+  tap,
+  withLatestFrom,
+  observeOn,
+} from 'rxjs/operators';
+import {
+  getKeyCodeValue,
+  getSliderPositionBasedOnValue,
+  getSliderValueForCoordinate,
+  roundToSnap,
+} from './slider-utils';
+
+/** We need unique ids in order to have correct labeling. */
+let uniqueId = 0;
+
+const KEY_CODES_ARRAY: Array<number> = [
+  LEFT_ARROW,
+  DOWN_ARROW,
+  RIGHT_ARROW,
+  UP_ARROW,
+  HOME,
+  END,
+  PAGE_UP,
+  PAGE_DOWN,
+];
+
+declare const window: any;
+
+@Component({
+  selector: 'dt-slider',
+  templateUrl: 'slider.html',
+  styleUrls: ['slider.scss'],
+  host: {
+    class: 'dt-slider',
+  },
+  encapsulation: ViewEncapsulation.Emulated,
+  exportAs: 'dtSlider',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DtSlider implements AfterViewInit, OnDestroy, OnInit {
+  /**
+   * Holds the value, the rounding is shifted with.
+   * Calculated based in the step, to avoid JS rounding problems.
+   */
+  private _roundShift: number = 0;
+
+  /** @internal Unique id for this input. */
+  _labelUid = `dt-slider-label-${uniqueId++}`;
+
+  /** Holds the value of the slider. */
+  private _value$ = new BehaviorSubject<number>(30);
+  /** Holds the description of the size of the slider. */
+  private _clientRect$: Observable<ClientRect>;
+  /** Observer that gets triggered if the slider is resized on the screen. */
+  private _resizeObserver$ = new Subject();
+  /** Variable to hold the ResizeObserver */
+  private _observer: any;
+  /** Observer that gets triggered if the input field value is changed. */
+  private inputFieldValue$ = new Subject<number>();
```

Comment:
Privates and internals are prefixed with an underscore

```suggestion
  private _inputFieldValue$ = new Subject<number>();
```

Reply:
I'll update this.